src/AudioThread.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+#include "PowerFSM.h"
+#include "concurrency/OSThread.h"
+#include "configuration.h"
+#include "main.h"
+#include "sleep.h"
+
+#ifdef HAS_I2S
+#include <AudioFileSourcePROGMEM.h>
+#include <AudioGeneratorRTTTL.h>
+#include <AudioOutputI2S.h>
+#include <ESP8266SAM.h>
+
+#define AUDIO_THREAD_INTERVAL_MS 25
+
+class AudioThread : public concurrency::OSThread
+{
+  public:
+    AudioThread() : OSThread("AudioThread") { initOutput(); }
+
+    void beginRttl(const void *data, uint32_t len)
+    {
+        setCPUFast(true);
```

Comment:
shouldn't the CPU clock be restored afterwards? Otherwise it consumes more (battery) power.